### PR TITLE
Calculating

### PR DESCRIPTION
**Pull request by tsohleDev**
**Connecting the components**

- Use props to send the calculation string from the parent state
- lifted the state from the buttons in order to make the calculations and send them to the screen

### CODE REVIEW

File: src/Components/Calculator.js [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
+    const { sum } = this.state;
+    const { total, next, operation } = sum;
     return (
       <div className="calculator">
-        <Screen className="screen" />
-        {buttons.map((button) => <Button key={button.id} text={button} />)}
+        <Screen total={total} next={next} operation={operation} className="screen" />
+        {buttons.map((button) => (
+          <Button
+            key={button.id}
```
Comment (codepantha) [inline src/Components/Calculator.js]: - [x] The `id` attribute does not exist on the button item. This is why you have this console error asking for a "unique" key prop shown below 👇 Kindly do `key={button}` instead 👍 
image

File: src/Components/Screen.js [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
       </div>
     );
   }
 }
 
+Screen.propTypes = {
+  total: PropTypes.string.isRequired,
+  next: PropTypes.string.isRequired,
+  operation: PropTypes.string.isRequired,
+};
```
Comment (codepantha) [inline src/Components/Screen.js]: - [x] There are about 3 console errors which I'll show below 👇 as a result of your PropTypes.
image
1. the `total` prop should be a number and not a string, so please specify `PropTypes.number.isRequired`
2. when the page first loads and you have not begun any arithmetic operation, the value of `next` prop is `null` so specifying in your PropTypes that `next` is required will be problematic. Please specify `PropTypes.string` only.
3.  same as number 2 above. Please specify `operation` prop to be `PropTypes.string` only

Having done that, you will notice that you need a `defaultProps` declaration. Kindly do this 👇 
```javascript
Screen.defaultProps = {
  next: null,
  operation: null,
};
```
Please visit this [link](https://www.geeksforgeeks.org/reactjs-defaultprops/) if you need to research more on React defaultProps 😎

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @tsohleDev 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added events and functionality to calculator app 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @tsohleDev, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**